### PR TITLE
#2809 added statistics-service to core repo

### LINKED ISSUE #2809: Moving the statistics-service to Keptn API endpoint
Based on user requests, the statistics-service which is developed in keptn/sandbox should move closer to Keptn due to its useful features in providing a comprehensive usage statistic. Hence, it should become a part of Keptn.

**Tasks:**
Move microservice from sandbox to keptn/keptn (core)
 - Integration into the install process
 - Integration into build/test pipelines (push to DockerHub repo) 
 - Check development standards as applied for other services  
 - Expose the endpoint of the service on the Keptn API
 - Update of API documentation
 - Check resource footprint: Use monitoring data to verify the memory consumption of this service > update the resource allocation in the deployment manifest accordingly.
 - The CLI remains on *keptn/sandbox*

Definition of Done: 
- [ ] Service is part of the build/test pipeline
- [ ] Keptn API: `v1/statistics` exposes the endpoints of the statistics-service

### PR DESCRIPTION
**Pull request by bacherfl**
Closes #2809

Original PR: https://github.com/keptn/keptn/pull/2904 (was closed due to problems with merging/rebasing)

The resource consumption has been investigated in detail here: [Resource consumption details](https://github.com/keptn-sandbox/statistics-service/pull/11)
Signed-off-by: Florian Bacher <florian.bacher@dynatrace.com>

### CODE REVIEW

File: installer/manifests/keptn/charts/control-plane/templates/api-gateway-nginx.yaml [hunk 1]
```diff
@@ -148,6 +148,37 @@ data:
       proxy_set_header X-Forwarded-Proto $scheme;
     }
 
+    location {{ .Values.prefixPath }}/api/statistics/swagger-ui/swagger.yaml {
+      # auth via backend (if the subrequest returns a 2xx response code, the access is allowed. If it returns 401 or 403,
+      # the access is denied) before we store the file
+      # see http://nginx.org/en/docs/http/ngx_http_auth_request_module.
```
Comment (christian-kreuzberger-dtx) [inline installer/manifests/keptn/charts/control-plane/templates/api-gateway-nginx.yaml:154]: I think this is missing
```
auth_request               {{ .Values.prefixPath }}/api/v1/auth;
```

File: statistics-service/README.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+kubectl apply -f deploy/service_keptn_080.yaml -n keptn
+```
+
+## Delete in your Kubernetes cluster
+
+To delete a deployed *statistics-service*, use `deploy/service.yaml` from this repository and delete the Kubernetes resources:
+
+```console
+kubectl delete -f deploy/service.yaml -n keptn
+```
```
Comment (christian-kreuzberger-dtx) [inline statistics-service/README.md:40]: The marked lines are not needed, please remove from README

File: statistics-service/README.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+## How to use the service
+
+Once the service is deployed in your cluster, you can access it using `port-forward`:
+
+```
+kubectl port-forward -n keptn svc/statistics-service 8080
+``` 
+
```
Comment (christian-kreuzberger-dtx) [inline statistics-service/README.md:65]: port-forward is no longer needed, use Keptn API Calls instead

File: statistics-service/README.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+You can then browse the API docs at by opening the Swagger docs in your [browser](http://localhost:8080/swagger-ui/index.html).
+
+To retrieve usage statistics for a certain time frame, you need to provide the [Unix timestamps](https://www.epochconverter.com/) for the start and end of the time frame.
+E.g.:
+
+```
+http://localhost:8080/v1/statistics?from=1600656105&to=1600696105
```
Comment (christian-kreuzberger-dtx) [inline statistics-service/README.md:72]: Change to an URL via Keptn API

File: statistics-service/README.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+```
+
+cURL Example:
+
+```
+curl -X GET "http://localhost:8080/v1/statistics?from=1600656105&to=1600696105" -H "accept: application/json"
```
Comment (christian-kreuzberger-dtx) [inline statistics-service/README.md:78]: Please provide an example via Keptn API

**Review by christian-kreuzberger-dtx — CHANGES_REQUESTED** [review]
Only have some simple comments in README, everything else LGTM

**Review by bacherfl — COMMENTED** [review]
-

**Review by christian-kreuzberger-dtx — DISMISSED** [review]
LGTM

**Review by christian-kreuzberger-dtx — APPROVED** [review]
LGTM

**Pull request merged**